zeq/publisher.h

```diff
@@ -17,17 +19,70 @@ namespace detail { class Publisher; }
 /**
  * Serves and publishes events, consumed by Subscriber.
  *
+ * The session is tied to ZeroConf announcement and can be disabled with
+ * zeq::NO_ANNOUNCE_SESSION.
+ *
  * Example: @include tests/publisher.cpp
  */
 class Publisher
 {
 public:
     /**
-     * Create a publisher on the given URI.
+     * Create a default publisher.
+     *
+     * Constrains:
+     * - bound to all network interfaces
+     * - runs on a random port
+     * - announces itself on the _zeroeq_pub._tcp ZeroConf service as host:port
+     * - announces session \<username\> or ZEROEQ_SESSION from environment if set
+     *
+     * @throw std::runtime_error if session is empty or socket setup fails
+     */
+    ZEQ_API Publisher();
+
+    /**
+     * Create a publisher which announces the specified session.
+     *
+     * Constrains:
+     * - bound to all network interfaces
+     * - runs on a random port
+     * - announces itself on the _zeroeq_pub._tcp ZeroConf service as host:port
+     * - announces given session or ZEROEQ_SESSION from environment if set
```

Comment:
Shouldn't the explicit session provided on the constructor override ZEROEQ_SESSION?